### PR TITLE
chore(built): adding gulp prepublish and gulp pretest tasks

### PR DESCRIPTION
**Pull request by cnishina**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,9 +18,11 @@
     "jasmine": "2.4.1",
     "saucelabs": "~1.0.1",
     "glob": "~6.0",
+    "gulp": "^3.9.1",
```
Comment (juliemr) [inline package.json]: This should be in `devDependencies`

File: package.json [hunk 2]
```diff
@@ -18,9 +18,11 @@
...
     "adm-zip": "0.4.7",
     "optimist": "~0.6.0",
     "q": "1.4.1",
+    "run-sequence": "^1.1.5",
```
Comment (juliemr) [inline package.json]: Ditto

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,4 @@
+built/
 node_modules/
 selenium/
 testapp/inbrowsertest/
```
Comment (juliemr) [inline .gitignore]: I think we should we add lib/ to `.npmignore` as well, right? Or will TypeScript users need that for typing info?

File: gulpfile.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    console.log(child.stdout.toString());
+  }
+  if (child.status !== 0) {
+    throw new Error('jslint: child error');
+  }
+});
+
+gulp.task('pretest', function() {
+  runSequence(
+    ['webdriver:update', 'jslint']
```
Comment (juliemr) [inline gulpfile.js]: Should we also run `built:copy` here since we'll need it for testing now?

File: .gitignore (hunk 3 above)
Comment (cnishina) [inline .gitignore]: Done. Added `lib/` to the `.npmignore`

**Comment by juliemr** [comment]
Looks good! Go ahead and merge.

**Pull request merged**